File: backend/src/main/java/com/mini/credit/dto/credit/CreditEnCoursResponse.java
package com.mini.credit.dto.credit;

import com.mini.credit.enums.StatutCredit;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Builder
public class CreditEnCoursResponse {
    private Long id;
    private String numeroCredit;
    private String membreNomComplet;
    private BigDecimal montantAccorde;
    private BigDecimal totalPaye;
    private BigDecimal resteAPayer;
    private LocalDate dateDecaissement;
    private Integer dureeValeur;
    private String dureeUnite;
    private StatutCredit statut;
    private String antenneNom;
    private String agentTerrainNom;
    private String gestionnaireNom;
    private String controleurNom;
    private String chefBureauNom;
    private String caissierNom;
    private String devise;
}

File: frontend/src/app/features/credit/models/credit-en-cours-response.ts
import { StatutCredit } from './enums/statut-credit.enum';

export interface CreditEnCoursResponse {
  id: number;
  numeroCredit: string;
  membreNomComplet?: string | null;
  montantAccorde: number;
  totalPaye: number;
  resteAPayer: number;
  dateDecaissement?: string | null;
  dureeValeur?: number | null;
  dureeUnite?: string | null;
  statut: StatutCredit;
  antenneNom?: string | null;
  agentTerrainNom?: string | null;
  gestionnaireNom?: string | null;
  controleurNom?: string | null;
  chefBureauNom?: string | null;
  caissierNom?: string | null;
  devise?: string | null;
}
